### PR TITLE
[bitnami/spring-cloud-dataflow] make upgrade possible without rabbitmq deployed

### LINKED ISSUE #5781: [bitnami/spring-cloud-dataflow] make upgrade possible without rabbitmq deployed
**Which chart**:
spring-cloud-dataflow (2.8.0)

**Describe the bug**
Upgrading the chart when initially installed with "rabbitmq.enabled=false" leads to the following Error:
```
Error: UPGRADE FAILED: template: spring-cloud-dataflow/templates/NOTES.txt:88:4: executing "spring-cloud-dataflow/templates/NOTES.txt" at <include "common.errors.upgrade.passwords.empty" (dict "validationErrors" $passwordErrors "context" $)>: error calling include: template: spring-cloud-dataflow/charts/common/templates/_errors.tpl:18:48: executing "common.errors.upgrade.passwords.empty" at <fail>: error calling fail:
PASSWORDS ERROR: you must provide your current passwords when upgrade the release
    'mariadb.auth.rootPassword' must not be empty, please add '--set mariadb.auth.rootPassword=$MARIADB_ROOT_PASSWORD' to the command. To get the current value:

        export MARIADB_ROOT_PASSWORD=$(kubectl get secret --namespace "scdf" scdf-mariadb -o jsonpath="{.data.mariadb-root-password}" | base64 --decode)

    'rabbitmq.auth.password' must not be empty, please add '--set rabbitmq.auth.password=$RABBITMQ_PASSWORD' to the command. To get the current value:

        export RABBITMQ_PASSWORD=$(kubectl get secret --namespace "scdf" scdf-rabbitmq -o jsonpath="{.data.rabbitmq-password}" | base64 --decode)

    'rabbitmq.auth.erlangCookie' must not be empty, please add '--set rabbitmq.auth.erlangCookie=$RABBITMQ_ERLANG_COOKIE' to the command. To get the current value:

        export RABBITMQ_ERLANG_COOKIE=$(kubectl get secret --namespace "scdf" scdf-rabbitmq -o jsonpath="{.data.rabbitmq-erlang-cookie}" | base64 --decode)
```

**To Reproduce**

1. install the chart with "rabbitmq.enabled=false"
  e.g. `helm install scdf . --set rabbitmq.enabled=false`
2. change anything and do a "helm upgrade"
  e.g. `helm upgrade scdf . --set rabbitmq.enabled=false --set server.livenessProbe.enabled=false`
4. upgrade will not succeed and post the error

**Expected behavior**
A helm upgrade should be possible, even when there is no rabbitmq used.

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.5.2", GitCommit:"167aac70832d3a384f65f9745335e9fb40169dc2", GitTreeState:"dirty", GoVersion:"go1.15.7"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1", GitCommit:"c4d752765b3bbac2237bf87cf0b1c2e307844666", GitTreeState:"clean", BuildDate:"2020-12-18T12:09:25Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.2-34+350770ed07a558", GitCommit:"350770ed07a558ba1ce8202a6f9ea849f9a75f06", GitTreeState:"clean", BuildDate:"2021-02-11T20:10:29Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by brjos**
Signed-off-by: Bruno Jost <bruno.jost@bedag.ch>

**Description of the change**

fixes the bug: [[bitnami/spring-cloud-dataflow] make upgrade possible without rabbitmq deployed](https://github.com/bitnami/charts/issues/5781)

**Benefits**

Allow to do 'helm upgrade', even when there is no rabbitmq deployed.

**Possible drawbacks**

**Applicable issues**

  - fixes #5781

**Additional information**

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/spring-cloud-dataflow/templates/NOTES.txt [hunk 1]
```diff
@@ -94,7 +94,7 @@ To access Spring Cloud Data Flow dashboard from outside the cluster execute the
     {{- $requiredExternalDbPassword := dict "valueKey" "externalDatabase.password" "secret" $secretNameExternalDb "field" "mariadb-password" -}}
     {{- $passwordWarnings =  append $passwordWarnings $requiredExternalDbPassword -}}
   {{- end -}}
-  {{- if and (.Values.externalRabbitmq.enabled) (not .Values.rabbitmq.enabled) (not .Values.externalRabbitmq.existingPasswordSecret) -}}
```
Comment (alemorcuq) [inline bitnami/spring-cloud-dataflow/templates/NOTES.txt:97]: How are you deploying the chart? For this block to execute you not only need to set `rabbitmq.enabled` to false (as you did), but also `externalRabbitmq.enabled` to true, which does not seem to be your case.

**Review by alemorcuq — CHANGES_REQUESTED** [review]
How are you deploying the chart? For this block to execute you not only need to

File: bitnami/spring-cloud-dataflow/templates/NOTES.txt (hunk 1 above)
Comment (brjos) [reply bitnami/spring-cloud-dataflow/templates/NOTES.txt:97]: externalRabbitmq is false in the [default values.yaml](https://github.com/bitnami/charts/blob/master/bitnami/spring-cloud-dataflow/values.yaml#L974) of the spring-cloud-dataflow chart.

Comment (alemorcuq) [reply bitnami/spring-cloud-dataflow/templates/NOTES.txt:97]: If you are using the default false value then this line is not affecting you as it requires `.Values.externalRabbitmq.enabled` to be true.

Comment (brjos) [reply bitnami/spring-cloud-dataflow/templates/NOTES.txt:97]: you are absolutely right! sorry. :(

Comment (alemorcuq) [reply bitnami/spring-cloud-dataflow/templates/NOTES.txt:97]: I'm going to close the PR. Let's try to identify the issue at #5781

**Pull request closed without merging**